tests/test_slurm_command_gen_strategy.py

```diff
@@ -94,3 +97,51 @@ def test_docker_image_url_is_file(self, nemo_cmd_gen: NeMoLauncherSlurmCommandGe
         Path(nemo_cmd_gen.final_cmd_args["docker_image_url"]).touch()
         nemo_cmd_gen.set_container_arg()
         assert nemo_cmd_gen.final_cmd_args["container"] == nemo_cmd_gen.final_cmd_args["docker_image_url"]
+
+
+class TestNeMoLauncherSlurmCommandGenStrategy__GenExecCommand:
+    @pytest.fixture
+    def nemo_cmd_gen(self, slurm_system: SlurmSystem) -> NeMoLauncherSlurmCommandGenStrategy:
+        env_vars = {"TEST_VAR": "VALUE"}
+        cmd_args = {"test_arg": "test_value"}
+        strategy = NeMoLauncherSlurmCommandGenStrategy(slurm_system, env_vars, cmd_args)
+        return strategy
+
+    def test_raises_if_required_env_var_missed(self, nemo_cmd_gen: NeMoLauncherSlurmCommandGenStrategy):
+        with pytest.raises(KeyError) as exc_info:
+            nemo_cmd_gen.gen_exec_command(
+                env_vars={}, cmd_args={}, extra_env_vars={}, extra_cmd_args="", output_path="", nodes=[]
+            )
+        assert REQUIRE_ENV_VARS[0] in str(exc_info.value)
+
+    def test_genv_vars_added(self, nemo_cmd_gen: NeMoLauncherSlurmCommandGenStrategy):
```

Comment:
What do you mean by 'genv_vars'? Do you mean general environment variables? It seems that you are testing additional environment variables in this case.

Reply:
I've renamed it to `test_extra_env_vars_added`. Is it better now?

Reply:
Sure, it looks good to me. Thanks.